jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java

```diff
@@ -405,6 +475,37 @@ public void destroyServer() throws Exception
         _server.join();
     }
 
+    @Test
+    public void testDestroyOrder() throws Exception
```

Comment:
Does a servlet added via the dynamic `ServletContext.addServlet()` methods during servlet init trip this up in any way?

Reply:
No,  ServletContext.addServlet() impl always creates a ServletHolder and adds it to the ServletHandler list of ServletHolders. The ServletHandler.initialize() method isn't called during the start sequence until AFTER the ServletContainerInitializers have already been called (and thus any ServletHolders created by the SCI will already have been added to the ServletHandler).